drivers/nrf_radio_802154/src/nrf_802154_trx.c

```diff
@@ -274,8 +274,8 @@ static void irq_init(void)
 
 #if NRF_802154_INTERNAL_RADIO_IRQ_HANDLING
     nrf_802154_irq_init(RADIO_IRQn, NRF_802154_IRQ_PRIORITY, nrf_802154_radio_irq_handler);
-    nrf_802154_irq_enable(RADIO_IRQn);
 #endif
+    nrf_802154_irq_enable(RADIO_IRQn);
```

Comment:
What's the purpose of this change? In what scenario external IRQ handler does not enable radio IRQ?

Reply:
I've discussed my question with @jciupis offline. I agree this function should be called out of `#if NRF_802154_INTERNAL_RADIO_IRQ_HANDLING` block. However, it looks that block between 271 and 277 should be called in `nrf_802154_trx_init()`, while only `nrf_802154_irq_enable()` should be called from `nrf_802154_trx_enable()`.

What do you think about such code reorganization?
Of course we can do that later, in scope of another clean-up task.